source/extensions/common/http_wasm/guest_config.h

```diff
@@ -0,0 +1,94 @@
+#pragma once
+
+#include <memory>
+
+#include "envoy/config/core/v3/base.pb.h"
+#include "envoy/extensions/filters/http/http_wasm/v3/wasm.pb.validate.h"
+#include "envoy/local_info/local_info.h"
+
+#include "source/common/protobuf/protobuf.h"
+#include "source/common/protobuf/utility.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace HttpWasm {
+
+//clang-format off
+using EnvironmentVariableMap = std::unordered_map<std::string, std::string>;
+struct SanitizationConfig {
+  std::vector<std::string> argument_list;
+  bool is_allowlist;
+};
+using AllowedCapabilitiesMap = std::unordered_map<std::string, SanitizationConfig>;
+// clang-format on
+
+class Context;
+class GuestConfig {
+public:
+  GuestConfig(const envoy::extensions::filters::http::http_wasm::v3::GuestConfig& config);
+  envoy::extensions::filters::http::http_wasm::v3::GuestConfig& config() { return config_; }
+  AllowedCapabilitiesMap& allowedCapabilities() { return allowed_capabilities_; }
+  EnvironmentVariableMap& environmentVariables() { return envs_; }
+
+private:
+  envoy::extensions::filters::http::http_wasm::v3::GuestConfig config_;
+  AllowedCapabilitiesMap allowed_capabilities_{};
+  EnvironmentVariableMap envs_;
+};
+
+using GuestConfigPtr = std::unique_ptr<GuestConfig>;
+
+// InitializedGuest contains the information for a filter/service.
+class InitializedGuest {
```

Comment:
how is this different from the mere `Guest` object? Would it be possible to consolidate them for simplicity?

Reply:
sorry for bad naming. Fixed/Renamed it to GuestConfig